src/modules/c.rs

```diff
@@ -0,0 +1,220 @@
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::c::CConfig;
+use crate::formatter::StringFormatter;
+use crate::formatter::VersionFormatter;
+
+use once_cell::sync::Lazy;
+use std::borrow::Cow;
+use std::ops::Deref;
+
+/// Creates a module with the current c version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("c");
+    let config: CConfig = CConfig::try_load(module.config);
+    let is_c_project = context
+        .try_begin_scan()?
+        .set_extensions(&config.detect_extensions)
+        .set_files(&config.detect_files)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_c_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        let c_compiler_info = Lazy::new(|| {
+            context.exec_cmds_return_first(&[
+                // the compiler is usually cc, and --version works on gcc and clang
+                &["cc", "--version"],
+                // but on some platforms gcc is installed as *gcc*, not cc
+                &["gcc", "--version"],
+                // for completeness, although I've never seen a clang that wasn't cc
+                &["clang", "--version"],
+            ])
+        });
+
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "compiler_name" => {
+                    let c_compiler_info = &c_compiler_info.deref().as_ref()?.stdout;
+
+                    let c_compiler = if c_compiler_info.contains("clang") {
+                        "clang"
+                    } else if c_compiler_info.contains("Free Software Foundation") {
+                        "gcc"
+                    } else {
+                        "Unknown compiler"
+                    };
+                    Some(c_compiler).map(Cow::Borrowed).map(Ok)
+                }
+                "compiler_version" => {
+                    let c_compiler_info = &c_compiler_info.deref().as_ref()?.stdout;
+                    if config.format.contains("$compiler_version")
```

Comment:
Please remove this line. For instance, `${compiler_version}` is also valid, but would be skipped by this code. If `$compiler_version` is matched in the `match`, it is in the format string.

Reply:
Ah, that's a leftover from my previous code that was looking for `$compiler_name` as well.